src/interpreter/index.ts

```diff
@@ -289,9 +292,9 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
                     return BrsInvalid.Instance;
                 }
             case Lexeme.Greater:
-                if (left.kind === ValueKind.Array || right.kind === ValueKind.Array) {
+                if (!isBrsPrimitive(left) || !isBrsPrimitive(right)) {
```

Comment:
`isBrsPrimitive` takes advantage of a TypeScript feature called ["type guards"](https://basarat.gitbooks.io/typescript/content/docs/types/typeGuard.html#user-defined-type-guards) &mdash; basically a function that returns a `boolean` but provides further information to the compiler.  Inside these `if` statements, the compiler is able to _guarantee_ that we only operate on properties and methods that exist on `BrsInvalid`, `BrsBoolean`, `BrsString`, `Int32`, `Int64`, `Float`, **and** `Double`.  TypeScript doesn't currently support user-defined type guards across multiple arguments though, so we're stuck with a "one at a time" kind of thing.

While this could be written as a single function call, I've chosen to leave it as is for a few reasons:

1. The compile-time guarantees that a BrightScript array or associative array won't make their way into this `if` block
2. The ability to compare a value is determined _in isolation_ of the other value, i.e. the LHS and RHS can vary independently.  While they will both affect the output, a failure in either immediately disallows comparison.  This feels like a clear way to describe that.